mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -855,18 +866,38 @@ static void populateDefaults() {
 }
 
 static LogicalResult detectMissingArguments() {
-  const static std::vector<const llvm::cl::opt<int64_t> *> requiredConvArgs = {
+  using RequiredArgsType = std::vector<const llvm::cl::opt<int64_t> *>;
+  static RequiredArgsType requiredConvArgs = {
```

Comment:
I think we can keep the const?

Reply:
I also prefer `const` but I am struggling to bind it to

https://github.com/ROCmSoftwarePlatform/rocMLIR/blob/636f6e72c5abd272502f8264d87838b5f08badd6/mlir/tools/rocmlir-gen/rocmlir-gen.cpp#L880-L885

It is not allowed to bind `const ref` to `ref`. I though it would be ugly to use `const_cast`; `ternary operator` would result in a lengthy expr.

Reply:
Did something like that

https://github.com/ROCmSoftwarePlatform/rocMLIR/blob/35480b33dcddceca4cb284bc0ad2d735fa2e28a1/mlir/tools/rocmlir-gen/rocmlir-gen.cpp#L877-L900